src/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs

```diff
@@ -115,6 +116,23 @@ public HttpResponseHeaders Headers
             }
         }
 
+        public HttpResponseHeaders TrailingHeaders
+        {
+            get
+            {
+                // Make sure that we don't inadvertently consuming trailing headers
+                // while draining a connection that's being returned back to the pool.
+                CheckDisposed();
```

Comment:
This doesn't seem like the right place to mitigate the draining scenario. I'd expect a drain check on the if statement before you parse the headers:
https://github.com/dotnet/corefx/pull/35337/files#diff-56140d22bd68e4555d62f331bfff602aR314

Reply:
Agreed, this comment and check should be removed.